app/src/components/acronyms.js

```diff
@@ -26,6 +27,64 @@ const acronyms = {
     }
   },
 
+  /**
+   *  @function getUsers
+   *  Fetch a specific acronym's application detail from GETOK database.
+   *  @param {string} applicationAcronym - The app specifier.
+   *  @returns {array} An array of acronym/user mappings. [] if none found
+   */
+  getUsers: async applicationAcronym => {
+    if (!applicationAcronym) {
+      const errMsg = 'No app acronym supplied to getUsers';
+      log.error('getUsers', errMsg);
+      throw new Error(errMsg);
+    }
+    try {
+      // Get the UserAcronym mappings from the DB
+      const acronymUsers = await acronymService.acronymUserList(applicationAcronym);
+      if (!acronymUsers || !acronymUsers.length) {
+        return [];
+      }
+      log.verbose('getUsers - acronymUsers', JSON.stringify(acronymUsers, 0, 2));
+
+      // Get the keycloak userinfo for the relevant users
+      // Could call GET /{realm}/users/{id} for each ID but more efficient to just get all users and filter server side here
+      const userInfos = await userComponent.getAllGetokUsers();
+      if (!userInfos || !userInfos.length) {
+        log.error('getUsers', `No users found in KC, something went wrong. ${applicationAcronym}, ${userInfos}`);
+        throw new Error(`An error occured fetching users for acronym ${applicationAcronym}`);
+      }
+
+      // Build the response object we want from the endpoint, joining the info from the table and the userInfo from KC
+      const filtered = acronymUsers.filter(au => userInfos.find(ui => ui.id === au.keycloakGuid));
+      const users = filtered.map(au => {
+        const userAcronymDetails = {
+          acronym: applicationAcronym,
+          owner: au.userAcronym.owner,
+          createdAt: au.userAcronym.createdAt
+        };
+        const kcUsr = userInfos.find(ui => ui.id === au.keycloakGuid);
+        return {
+          userAcronymDetails: userAcronymDetails,
+          user: {
+            userId: au.userAcronym.userId,
+            keycloakGuid: kcUsr.id,
+            username: kcUsr.username,
+            firstName: kcUsr.firstName,
+            lastName: kcUsr.lastName,
+            email: kcUsr.email
+          }
+        };
```

Comment:
Could in-line userAcronymDetails as part of the return object statement and save a const declaration.

Reply:
Yeah good point, looks a little cleaner too